### ISSUE TITLE
EEG: multiple dataset files can be submitted with different extension

### ISSUE THREAD

**Issue by arnodelorme** [bids-standard/bids-validator#1900]
This dataset contains both .set and .vhdr files 

https://nemar.org/dataexplorer/detail?dataset_id=ds003190

This should not be possible. There should be either one or the other.

**Comment by Remi-Gau** [comment]
Relates to https://github.com/bids-standard/bids-specification/issues/1487

Unless I am mistaken enforcing this would depend on updating the schema (see https://github.com/bids-standard/bids-specification/pull/1492) for the deno based validator.

I doubt the legacy validator will enforce this.

**Comment by sappelhoff** [comment]
I don't think we have a rule in BIDS that the data format chosen for a dataset must be consistent :thinking: 

although it **is**  a bit weird to mix them in a single dataset.

**Comment by effigies** [comment]
We do check whether someone has both `.nii` and `.nii.gz` files, as this is a somewhat common issue.

https://github.com/bids-standard/bids-validator/blob/932c782a556afea34346e994f655e03cf3e171fe/bids-validator/validators/nifti/duplicateFiles.js#L3-L34

A similar thing could be written for other formats, although it would be more complicated, because EEG has multi-file formats and files sharing the same stem isn't in itself an error.

**Comment by sappelhoff** [comment]
> We do check whether someone has both .nii and .nii.gz files, as this is a somewhat common issue.

but does it say in the spec, that these shouldn't be mixed?

**Comment by Remi-Gau** [comment]
I think we had added this section in the spec, no?

https://bids-specification.readthedocs.io/en/stable/common-principles.html#uniqueness-of-data-files

But I need to get back to finishing the schema pr.

**Comment by sappelhoff** [comment]
> I think we had added this section in the spec, no?

what we added there was:

- there MUST NOT be `data_a.jpg` and `data_a.tif` in the same dataset

However, here we have `data_a.jpg` and `data_b.tif` as a situation 🤔
